plugins/scaffolder-backend/src/scaffolder/actions/builtin/fetch/helpers.ts

```diff
@@ -64,10 +64,12 @@ export async function fetchContents({
         throw new InputError(`No integration found for location ${baseUrl}`);
       }
 
-      readUrl = integration.resolveUrl({
+      const resolvedUrl = integration.resolveUrl({
         url: fetchUrl,
         base: baseUrl,
       });
+
+      readUrl = decodeURIComponent(resolvedUrl);
```

Comment:
Feels super risky and not really the place to do this tbh. Referring to #11988, I actually feel that the issue is that the absolute URL is not encoded, rather than the relative URL being encoded. It's behavior that's built into `URL` and not something I think we should mess with.

For fixing the actual issue I think what we want to do is most likely to have `getBitbucketServerDownloadUrl` decode the `filepath` before it encodes it again. That way we avoid the duplicate encoding but also make sure that the path is always properly encoded.

Reply:
> I actually feel that the issue is that the absolute URL is not encoded, rather than the relative URL being encoded.

Yeah, that was also my first thought - I added a `new URL()` for the absolute URL, but it felt wrong to decode the URL in bitbucket and encode / prepare the URL beforehand.

So from a perspective of "separation of concerns" my thoughts were, that the (bitbucket) integration should handle everything with the URLs and that it shouldn't be encoded by anyone else beforehand. Otherwise you can never know who encodes the URL, if it is double / tripple encoded and if something it wanted / not wanted. So it should be the Bitbucket integration to take the raw value and do whatever it has to do with it.

But either way I am not really happy with the solutions, that's why I created the ticket and wanted to get your feedback! Do you have any architectural guidelines for topics like that?

Reply:
Small addition, why it felt wrong encoding the absolute URL as well - during the debugging it was quite hard to find "who is encoding the path". Took some time to find the line - once found it's obvious, but the way to it was quite bumpy.

Reply:
I think arguably `getBitbucketServerDownloadUrl` is the piece that's doing something wrong here though. It's encoding the `filepath` even though it might already be encoded. I say we need to decode it before encoding it again.

Reply:
@dschwank agree with @Rugvip here, do you want to make the changes so we can get this in? :pray:

Reply:
Sorry for not responding, I was sick for a couple of weeks. Thanks a lot for taking over and merging a fix @benjdlambert & @Rugvip!